### PR TITLE
Tp and test

### PR DESCRIPTION
**Pull request by nick-drozd**
The test commit adds a Python test file and a yaml network file. The latter has to be used with the current ST `run_docker_test` command, which isn't suitable. For now, it can be used for local testing.

### CODE REVIEW

**Comment by nick-drozd** [comment]
Note that the test uses actual real-life data -- for instance, the ISWC of David Bowie's song "Cat People" (or at least one of them; I don't really understand how it works) really is T-905.465.061-2. This could be converted to documentation.

File: omi/sawtooth_omi/handler.py [hunk 1]
```diff
@@ -72,11 +72,10 @@ def _get_unique_key(obj, action):
     return key
 
 
-def make_omi_address(obj, action):
+def make_omi_address(name, action):
```
Comment (peterschwarz) [inline omi/sawtooth_omi/handler.py]: tying this to action, instead of object type is still seems a bit off.

**Review by peterschwarz — COMMENTED** [review]
How do I build the test docker container?  I get when trying :
```
ERROR: repository sawtooth-tp_omi not found: does not exist or no pull access
```

File: omi/sawtooth_omi/handler.py (hunk 1 above)
Comment (nick-drozd) [reply omi/sawtooth_omi/handler.py]: Yeah, it's not great. I'll fix that after I get the rest of this functionality written.

**Comment by nick-drozd** [comment]
Instructions for testing (for now):

1) After running `protogen`, copy `omi` to `sawtooth-core/families` and copy `omi-tp` to `sawtooth-core/bin`

2) Add `tp_omi` to the mounted `build_python` section of `build_all`:
```
    build_python() {
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-dev-python
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-rest_api
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_config
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_omi # <----------
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_intkey_python
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_supplychain_python
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_validator_registry
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-tp_xo_python
        docker_build $top_dir/docker/sawtooth-dev-python docker/ sawtooth-validator
        docker_run sawtooth-dev-python
    }
```

3) Run `build_all -l python`.

4) Run `run_docker_test sawtooth-core/families/omi/tests/omi-smoke.yaml`.

**Comment by peterschwarz** [comment]
This needs to be updated to handle the new validator arguments:
```
Attaching to latest_validator_1, latest_tp_config_1, latest_rest_api_1, latest_tp_omi_1, latest_integration_test_1
validator_1         | writing file: /etc/sawtooth/keys/validator.priv
validator_1         | writing file: /etc/sawtooth/keys/validator.pub
validator_1         | Generating /var/lib/sawtooth/genesis.batch
validator_1         | usage: validator [-h] [--config-dir CONFIG_DIR] [-B BIND]
validator_1         |                  [-P {static,dynamic}] [-E ENDPOINT] [-s SEEDS] [-p PEERS]
validator_1         |                  [-v]
validator_1         | validator: error: unrecognized arguments: --public-uri tcp://validator:8800 --component-endpoint  tcp://eth0:40000 --network-endpoint tcp://eth0:8800
```

File: omi/tests/omi-smoke.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  tp_omi:
+    image: sawtooth-tp_omi:$ISOLATION_ID
+    volumes:
+      - $SAWTOOTH_CORE:/project/sawtooth-core
+    expose:
+      - 40000
+    depends_on:
+      - validator
+    command: tp_omi -vv tcp://validator:40000
```
Comment (rbuysse) [inline omi/tests/omi-smoke.yaml]: Should this be omi-tp?

**Comment by nick-drozd** [comment]
The yaml file has been updated.

**Review by peterschwarz — APPROVED** [review]

**Review by jsmitchell — APPROVED** [review]

**Pull request merged**